File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/ActivityType.java
package com.esferalia.aon.occam.api.model;

import java.io.Serializable;

public class ActivityType  implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private Integer id;
	private Integer domain;
	private String description;
	private Integer projectType;
	private boolean active;
	private boolean dirty;
	
	public Integer getId() {
		return id;
	}

	public ActivityType setId(Integer id) {
		this.id = id;
		return this;
	}

	public Integer getDomain() {
		return domain;
	}

	public ActivityType setDomain(Integer domain) {
		this.domain = domain;
		return this;
	}

	public String getDescription() {
		return description;
	}

	public ActivityType setDescription(String description) {
		setDirty(true);
		this.description = description;
		return this;
	}
	
	
	public Integer getProjectType() {
		return projectType;
	}

	public ActivityType setProjectType(Integer projectType) {
		setDirty(true);
		this.projectType = projectType;
		return this;
	}

	public boolean isActive() {
		return active;
	}
	
	public ActivityType setActive(boolean active) {
		setDirty(true);
		this.active = active;
		return this;
	}
	
	public boolean isEmpty() {
		return getId() == null && getDomain() == null
			&& getDescription() == null;
	}

	public boolean isDirty() {
		return dirty;
	}
	
	public ActivityType setDirty(boolean dirty) {
		this.dirty = dirty;
		return this;
	}
}

File: aon-solutions/src/main/webapp/models/ActivityType.js
import * as LS from '../services/localStorageService.js';

export class ActivityType {
    id;
    domain;
    description;
    projectType;
    active;
    dirty;

    constructor(activityType) {
        if(activityType) {
            this.id = activityType.id;
            this.domain = activityType.domain || LS.getDomainId();
            this.description = activityType.description || '';
            this.projectType = activityType.projectType || '';
            this.active = activityType.active || true;
            this.dirty = activityType.dirty || false;
        } else {
            this.domain = LS.getDomainId();
            this.description = '';
            this.active = true;
            this.dirty = false;
        }
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
        return this;
    }

    getDomain() {
        return this.domain;  
    }

    setDomain(domain) {
        this.domain = domain;
        return this;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.setDirty(true);
        this.description = description;
        return this;
    }

    getProjectType() {
        return this.projectType;
    }

    setProjectType(projectType) {
        this.setDirty(true);
        this.projectType = projectType;
        return this;
    }

    isActive() {
        return this.active;
    }

    setActive(active){
        this.setDirty(true);
        this.active = active;
        return this;
    }

    isDirty() {
        return this.dirty;
    }

    setDirty(dirty) {
        this.dirty = dirty;
        return this;
    }
} 